projects/online/online/make_event_pages.py

```diff
@@ -0,0 +1,194 @@
+    """
+    Embed an image in HTML using base64 encoding.
+
+    Args:
+        image_path: Path to the image file.
+        caption: Caption for the image.
+
+    Returns:
+        str: HTML string with the embedded image.
+    """
+    with open(image_path, "rb") as img_file:
+        encoded_string = base64.b64encode(img_file.read()).decode("utf-8")
+    return f'''
+        <div class="item">
+            <img src="data:image/png;base64,{encoded_string}" alt="{caption}">
+            <div class="caption">{caption}</div>
+        </div>
+    '''
+
+
+def generate_html(event: Path, outdir: Path):
+    """
+    Generate an HTML summary page for the event.
+
+    Args:
+        eventdir: Directory containing event data.
+        outdir: Output directory for the HTML file.
+    """
+    eventdir = outdir / event.stem
+    plotsdir = eventdir / "plots"
+    html_file = eventdir / f"{event.stem}.html"
+
+    with open(html_file, "w") as f:
+        f.write(html_header(event))
+        for png in plotsdir.glob("*.png"):
+            caption = png.stem
+            f.write(embed_image(png, caption))
+        f.write(html_footer())
+
+
+def process_event_outputs(event: Path, outdir: Path):
+    eventdir = outdir / event.stem
+    plotsdir = eventdir / "plots"
+    eventdir.mkdir(parents=True, exist_ok=True)
+    plotsdir.mkdir(parents=True, exist_ok=True)
+
+    with open(event / f"{event.stem}.json", "r") as f:
+        gpstime = json.load(f)["gpstime"]
+
+    for png in event.glob("*.png"):
+        shutil.copy(png, plotsdir / png.name)
+
+    with h5py.File(event / "output.hdf5", "r") as f:
+        time = f["time"][:]
+        output = f["output"][:]
+        integrate = f["integrated"][:]
+
+    plt.plot(time, output, label="Raw output")
+    plt.plot(time, integrate, label="Integrated output")
+    plt.axvline(gpstime, color="red", linestyle="--", label="Event time")
+    plt.xlabel("GPS time")
+    plt.ylabel("Detection statistic")
+    plt.legend()
+    plt.savefig(plotsdir / "aframe_response.png", dpi=150)
+    plt.close()
+
+    asds = np.load(event / "asd.npy")[0]
+    freqs = asds[0]
+    asds = asds[1:]
+
+    ifos = ["H1", "L1", "V1"]
+    for i, ifo in enumerate(ifos[: len(asds)]):
+        plt.plot(freqs, asds[i], label=ifo)
+    plt.xlabel("Frequency (Hz)")
+    plt.ylabel("ASD (strain/Hz^0.5)")
+    plt.xscale("log")
+    plt.yscale("log")
+    plt.legend()
+    plt.savefig(plotsdir / "asds.png", dpi=150)
+    plt.close()
+
+
+def main(event_dir: Path, outdir: Path):
+    if not outdir.exists():
+        outdir.mkdir(parents=True, exist_ok=True)
+    while True:
+        new_events = set(event_dir.iterdir()).difference(set(outdir.iterdir()))
+        if new_events:
+            for event in tqdm(new_events):
+                try:
+                    process_event_outputs(event, outdir)
+                    generate_html(event, outdir)
+                except FileNotFoundError:
```

Comment:
This exists so that the script doesn't break when it encounters events that are missing various outputs, either because the pipeline had an error, or because we added new output files.